front/components/agent_builder/triggers/TriggerViewsSheet.tsx

```diff
@@ -36,152 +56,336 @@ export function TriggerViewsSheet({
+          webhookSourceView={editWebhookSourceView ?? selectedWebhookSourceView}
+          isEditor={webhookIsEditor}
+        />
+      ),
+    },
+  ];
 
   return (
-    <>
-      <TriggerSelectionPage
-        isOpen={isSelectionSheetOpen}
-        onClose={handleSelectionSheetClose}
-        onScheduleSelect={handleScheduleSelect}
-        onWebhookSelect={handleWebhookSelect}
-        webhookSourceViews={webhookSourceViews}
-      />
-
-      <ScheduleEdition
-        owner={owner}
-        trigger={scheduleEditionState?.trigger ?? null}
-        isOpen={scheduleEditionState !== null}
-        onClose={handleScheduleClose}
-        onSave={handleScheduleSave}
-      />
-
-      <WebhookEdition
-        owner={owner}
-        trigger={webhookEditionState?.trigger ?? null}
-        isOpen={webhookEditionState !== null}
-        onClose={handleWebhookClose}
-        onSave={handleWebhookSave}
-        agentConfigurationId={agentConfigurationId}
-        webhookSourceView={webhookEditionState?.webhookSourceView ?? null}
-      />
-    </>
+    <FormProvider form={form} onSubmit={handleFormSubmit}>
+      <MultiPageSheet
+        open={mode !== null}
+        onOpenChange={(open) => !open && handleSheetClose()}
+      >
+        <MultiPageSheetContent
+          pages={pages}
+          currentPageId={currentPageId}
+          onPageChange={(pageId) => setCurrentPageId(pageId as PageId)}
+          size="xl"
+          addFooterSeparator
+          showHeaderNavigation={false}
+          showNavigation={false}
+          leftButton={
+            currentPageId !== TRIGGERS_SHEET_PAGE_IDS.SELECTION
+              ? {
+                  label: "Cancel",
+                  variant: "outline",
+                  onClick: handleCancel,
+                }
+              : {
+                  label: "Close",
+                  variant: "outline",
+                  onClick: handleSheetClose,
+                }
+          }
+          rightButton={
+            currentPageId === TRIGGERS_SHEET_PAGE_IDS.SCHEDULE
+              ? {
+                  label:
+                    editTrigger?.kind === "schedule"
+                      ? "Update Trigger"
+                      : "Add Trigger",
+                  variant: "primary",
+                  onClick: form.handleSubmit(handleFormSubmit),
+                }
+              : currentPageId === TRIGGERS_SHEET_PAGE_IDS.WEBHOOK
+                ? {
+                    label:
+                      editTrigger?.kind === "webhook"
+                        ? "Update Trigger"
+                        : editWebhookSourceView ?? selectedWebhookSourceView
+                          ? `Add ${(editWebhookSourceView ?? selectedWebhookSourceView)?.customName} Trigger`
+                          : "Add Trigger",
+                    variant: "primary",
+                    onClick: form.handleSubmit(handleFormSubmit),
+                  }
+                : undefined
```

Comment:
that's quite the ternary
lgtm but not the most readable, could we maybe export it to a function ?

feel free to discard if you already want to kill me for making you do front end ^^